roles/dev/tasks/setup_ssl.yml

```diff
@@ -0,0 +1,142 @@
+---
+# Transport Layer Security
+- name: Enable TLS support in Apache
+  become: yes
+  apache2_module: name=ssl state=present
+  when: setup_ssl
+  notify: restart apache
+
+- name: Setup self signed SSL certificate
+  shell: |
+    cd {{ROOTDIR}}
+    if [[ $* == *--ssl* ]]; then
+      bash {{SCRIPTDIR}}/cert_selfsigned.sh
```

Comment:
BTW I just stumbled upon this: https://www.jeffgeerling.com/blog/2017/self-signed-certificates-ansible-local-testing-nginx — might be helpful.